File: ScalaExamples/src/Arcade/Intro/EdgeOfTheOcean.scala
package Arcade.Intro

class EdgeOfTheOcean {

	def adjacentElementsProduct(inputArray: Array[Int]): Int = {
		var aux = 0
		var result = Int.MinValue

		for(i <- 0 until inputArray.length - 1) {
		  aux = inputArray(i) * inputArray(i + 1)
			if (aux > result) {
				result = aux
			}
		}

		result
	}

	def shapeArea(n: Int): Int = {
		var result = 1

		for(i <- 0 until n) {
		  result += i * 4
			println(result)
		}

		result
	}

	def makeArrayConsecutive2(statues: Array[Int]): Int = {
		statues.max - statues.min - statues.length + 1
	}

	def almostIncreasingSequence(sequence: Array[Int]): Boolean = {
		var count = 0
		
		for (i <- 0 until sequence.length - 1) {
			if (sequence(i) - sequence(i + 1) >= 0) {
				count += 1
				if (i - 1 >= 0 && i + 2 <= sequence.length - 1
					&& sequence(i) - sequence(i + 2) >= 0
					&& sequence(i - 1) - sequence(i + 1) >= 0) {
					return false
				}
			}
		}
		
		count <= 1
	}
	
	def matrixElementsSum(matrix: Array[Array[Int]]): Int = {
		var result = 0
		var zeros = Array[Int]()
		
		for (i <- matrix.indices) {
			for (j <- matrix(i).indices) {
				if (!zeros.contains(j)) {
					if (matrix(i)(j) == 0) {
						zeros = zeros :+ j
					} else {
						result += matrix(i)(j)
					}
				}
			}
		}
		
		result
	}

}

File: CodeProjects/Arcade/EdgeOfTheOcean.cs
using System;
using System.Linq;

namespace CodeProjects.Arcade
{
	class EdgeOfTheOcean
	{
		int adjacentElementsProduct(int[] inputArray)
		{
			int result = int.MinValue;
			int aux;

			for (int i = 0; i < inputArray.Length - 1; i++)
			{
				aux = inputArray[i] * inputArray[i + 1];
				if (aux > result)
				{
					result = aux;
				}
			}

			return result;
		}

		int shapeArea(int n)
		{
			int result = 1;

			for (int i = 0; i < n; i++)
			{
				result += i * 4;
			}

			return result;
		}

		int makeArrayConsecutive2(int[] statues)
		{
			int result = 0;
			int aux;
			int[] sortedStatues = statues.OrderBy(i => i).ToArray();

			for (int i = 1; i < sortedStatues.Length; i++)
			{
				aux = sortedStatues[i] - 1;
				while (sortedStatues[i - 1] != aux)
				{
					result++;
					aux--;
				}
			}

			return result;
		}

		bool almostIncreasingSequence(int[] sequence)
		{
			bool removedOne = false;
			int i = 0;

			while (i < sequence.Length - 1)
			{
				if (sequence[i] + 1 != sequence[i + 1])
				{
					if (!removedOne && i < sequence.Length - 2 && sequence[i] + 2 == sequence[i + 2])
					{
						i++;
					}
					else
					{
						return false;
					}
				}

				i++;
			}

			return true;
		}
	}
}
